src/protocol/vm/tycho_decoder.rs

```diff
@@ -133,25 +136,29 @@ impl TryFromWithBlock<ComponentWithState> for VMPoolState<PreCachedDB> {
         let adapter_file_path =
             format!("src/protocol/vm/assets/{}", to_adapter_file_name(protocol_name));
         info!("Creating a new pool state for balancer pool with id {}", &id);
-        let mut pool_state = VMPoolState::new(
-            id.clone(),
-            tokens.clone(),
-            block,
-            balances,
-            balance_owner,
-            adapter_file_path,
-            involved_contracts,
-            stateless_contracts,
-            manual_updates,
-            false,
-        )
-        .await
-        .map_err(InvalidSnapshotError::VMError)?;
+
+        let mut pool_state_builder = VMPoolStateBuilder::new(id.clone(), tokens.clone(), block)
+            .balances(balances)
```

Comment:
is this mandatory in `VMPoolState`? If yes it needs to be in the `new` function

Reply:
Yes I am talking about balances, not sure if I should include it in the mandatory fields or not